build.gradle

```diff
@@ -29,6 +29,21 @@ allprojects {
 
 apply plugin: 'java'
 
+/* SET PROJECT INFO for all artifacts */
+assert hasProperty('version')   // See version in gradle.properties.
+System.out.println("version=${version}")
```

Comment:
You are right, I did not take care of version well. Should do it using the build_version. 
Regarding to group, there is another question. I think the group for edgent is a little weird. It seems to me that a better setting is to unify all groups to org.apache.edgent instead of having different group ids across the project. But that may involve some additional work, plus I am not sure if there are special reasons for quarks to do so, I just leave it as is.